src/devices_models/devices/hydro_generation.jl

```diff
@@ -11,164 +11,95 @@ struct HydroCommitmentRunOfRiver <: AbstractHydroUnitCommitment end
 struct HydroCommitmentReservoirBudget <: AbstractHydroUnitCommitment end
 struct HydroCommitmentReservoirStorage <: AbstractHydroUnitCommitment end
 
-########################### Hydro generation variables #################################
+########################### ActivePowerVariable, HydroGen #################################
 
-"""
-This function add the variables for active power to the model
-"""
-function AddVariableSpec(
-    ::Type{T},
-    ::Type{U},
-    ::PSIContainer,
-) where {T <: ActivePowerVariable, U <: PSY.HydroGen}
-    return AddVariableSpec(;
-        variable_name = make_variable_name(T, U),
-        binary = false,
-        expression_name = :nodal_balance_active,
-        initial_value_func = x -> PSY.get_active_power(x),
-        lb_value_func = x -> PSY.get_active_power_limits(x).min,
-        ub_value_func = x -> PSY.get_active_power_limits(x).max,
-    )
-end
+get_variable_binary(::ActivePowerVariable, ::Type{<:PSY.HydroGen}) = false
+get_variable_expression_name(::ActivePowerVariable, ::Type{<:PSY.HydroGen}) = :nodal_balance_active
 
-"""
-This function add the variables for reactive power to the model
-"""
-function AddVariableSpec(
-    ::Type{T},
-    ::Type{U},
-    ::PSIContainer,
-) where {T <: ReactivePowerVariable, U <: PSY.HydroGen}
-    return AddVariableSpec(;
-        variable_name = make_variable_name(T, U),
-        binary = false,
-        expression_name = :nodal_balance_reactive,
-        initial_value_func = x -> PSY.get_reactive_power(x),
-        lb_value_func = x -> PSY.get_reactive_power_limits(x).min,
-        ub_value_func = x -> PSY.get_reactive_power_limits(x).max,
-    )
-end
+get_variable_initial_value(pv::ActivePowerVariable, d::PSY.HydroGen, settings) =
+    get_variable_initial_value(pv, d, get_warm_start(settings) ? WarmStartVariable() : ColdStartVariable())
+get_variable_initial_value(::ActivePowerVariable, d::PSY.HydroGen, ::WarmStartVariable) = PSY.get_active_power(d)
+get_variable_initial_value(::ActivePowerVariable, d::PSY.HydroGen, ::ColdStartVariable) = nothing
```

Comment:
we don't have coldstart for Hydro. This is only for thermals.

Reply:
Resolved in https://github.com/NREL-SIIP/PowerSimulations.jl/pull/560/commits/20ab12eff7f5adf899419a65ff2d2587a4cb9a95